sway-lib-std/src/i16.sw

```diff
@@ -0,0 +1,140 @@
+library i16;
+
+use core::num::*;
+use ::assert::assert;
+
+/// The 8-bit signed integer type.
+/// Represented as an underlying u16 value.
+pub struct i16 {
+    underlying: u16,
+}
+
+pub trait From {
+    /// Function for creating i16 from its u16 and bool components.
+    fn from(value: u16) -> Self;
+}
+
+impl From for i16 {
+    fn from(value: u16) -> i16 {
+        i16 {
+            underlying: value,
```

Comment:
Awesome implementation! A couple of notes here -- since we follow Rust generally when it comes to these kinds of things,  we should probably follow Rust's conventions on integer width conversions: https://doc.rust-lang.org/std/primitive.i32.html#impl-From%3Ci16%3E

Note how `i32` can only be constructed from `u16`, `u8`, `i16`, or `i8` because they have no risk of overflow. 

Additionally, check out this example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=849026d21ceb5f4632f8b179665f1e7d

I believe with your `from()` here, `i16::from(5u32)` would be `i16::min() + 5`, which is not consistent.

Reply:
The From trait in Sway is not the same, as in Rust. Basically we don't have one From trait (e.g. the for U128 differs significantly). From normally is for direct conctructing and should be used with caution.
About overflows: Sway seems to have an undesirable behavior to ignore them. E.g. you could use u32 instead of u16, and you wouldn't get any panics. I'd like it to be different, but I guess that's out of the scope of this PR